pkg/graveler/graveler.go

```diff
@@ -1474,6 +1475,17 @@ func (g *Graveler) Delete(ctx context.Context, repository *RepositoryRecord, bra
 		})
 }
 
+func (g *Graveler) deleteConsumer(ctx context.Context, cancel context.CancelFunc, repository *RepositoryRecord, branch *Branch, cachedMetaRangeID *MetaRangeID, keysChan <-chan Key) error {
+	for key := range keysChan {
+		err := g.deleteUnsafe(ctx, repository, branch, key, cachedMetaRangeID)
+		if err != nil {
+			cancel()
+			return err
```

Comment:
```suggestion
		if err != nil {
		if !errors.Is(ctx, context.Canceled) {
			cancel()
		}
		return err
```
Otherwise when _one_ error occurs, every call is cancelled.  If there are 100 workers then that's 100 cancellations; [the code](https://cs.opensource.google/go/go/+/refs/tags/go1.19.4:src/context/context.go;l=394-433;drc=db36eca33c389871b132ffb1a84fd534a349e8d8) is tricky for me to grasp, but I'm pretty sure there may be 99 mutex contentions simply due to some layer on the stack having to call ctx.Err to retrieve the cancellation error.